osxkeychain_test.go

```diff
@@ -234,3 +234,21 @@ func TestRemoveAndAddGenericPassword(t *testing.T) {
 		t.Error(err)
 	}
 }
+
+func TestAddGenericPasswordWithApplicationAccess(t *testing.T) {
+	serviceName := "osxkeychain_test"
+	accountName := "test account"
+	applicationPath := "/Applications/Mail.app"
+	data := []byte("testdata")
+
+	err := AddGenericPasswordWithApplication(serviceName, accountName, data, applicationPath)
+	if err != nil {
+		t.Error(err)
+	}
+
+	// Cleanup
+	err = FindAndRemoveGenericPassword(&GenericPasswordAttributes{ServiceName: serviceName, AccountName: accountName})
```

Comment:
Ah, I guess if you use the symmetric key type, you won't be able to use this function.

I think you can do something similar to GetAllAccountNames, though. You'd build a query dictionary with kSecClass = kSecClassSymmetricKey (or whatever), then call SecItemDelete on it.